packages/open-next/src/adapters/routing/matcher.ts

```diff
@@ -233,18 +232,12 @@ export function fixDataPage(internalEvent: InternalEvent, buildId: string) {
     let newPath = rawPath.replace(dataPattern, "").replace(/\.json$/, "");
     newPath = newPath === "/index" ? "/" : newPath;
     query.__nextDataReq = "1";
-    const urlQuery: Record<string, string> = {};
-    Object.keys(query).forEach((k) => {
-      const v = query[k];
-      urlQuery[k] = Array.isArray(v) ? v.join(",") : v;
-    });
+
     return {
       ...internalEvent,
       rawPath: newPath,
       query,
-      url: `${newPath}${
-        query ? `?${new URLSearchParams(urlQuery).toString()}` : ""
-      }`,
+      url: `${newPath}${query ? "?" + convertToQueryString(query) : ""}`,
```

Comment:
I think it might be cleaner to move the ternary logic to the `convertToQueryString` function itself.

`url.search = '?a=a'` would still work w/ the prefixed `?`